From ed69f90779df1a42c8e5db671502658f9bcf88fe Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:43:21 -0400
Subject: [ethernet] cxgb4: Decode the firmware port and module type a bit more for ethtool

Message-id: <1410475447-94392-63-git-send-email-svemuri@redhat.com>
Patchwork-id: 93812
O-Subject: [RHEL7.1 PATCH BZ 1124947 062/108] cxgb4: Decode the firmware port and module type a bit more for ethtool
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Signed-off-by: Hariprasad Shenai <hariprasad@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 3e00a5093561ac14a73e1dcfbaf0d9f26db65141)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
index b04ed15..ae32a74 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
@@ -2252,12 +2252,19 @@ static int get_settings(struct net_device *dev, struct ethtool_cmd *cmd)
  else if (p->port_type == FW_PORT_TYPE_FIBER_XFI ||
    p->port_type == FW_PORT_TYPE_FIBER_XAUI)
   cmd->port = PORT_FIBRE;
- else if (p->port_type == FW_PORT_TYPE_SFP) {
-  if (p->mod_type == FW_PORT_MOD_TYPE_TWINAX_PASSIVE ||
-      p->mod_type == FW_PORT_MOD_TYPE_TWINAX_ACTIVE)
+ else if (p->port_type == FW_PORT_TYPE_SFP ||
+   p->port_type == FW_PORT_TYPE_QSFP_10G ||
+   p->port_type == FW_PORT_TYPE_QSFP) {
+  if (p->mod_type == FW_PORT_MOD_TYPE_LR ||
+      p->mod_type == FW_PORT_MOD_TYPE_SR ||
+      p->mod_type == FW_PORT_MOD_TYPE_ER ||
+      p->mod_type == FW_PORT_MOD_TYPE_LRM)
+   cmd->port = PORT_FIBRE;
+  else if (p->mod_type == FW_PORT_MOD_TYPE_TWINAX_PASSIVE ||
+    p->mod_type == FW_PORT_MOD_TYPE_TWINAX_ACTIVE)
    cmd->port = PORT_DA;
   else
-   cmd->port = PORT_FIBRE;
+   cmd->port = PORT_OTHER;
  } else
   cmd->port = PORT_OTHER;
 
-- 
1.7.1